README.md

```diff
@@ -1,16 +1,329 @@
+
+## What is Mambajs?
+
+Mambajs is a powerful toolbox designed for manipulating conda environment definitions, with first-class support for:
+
+- **Conda package management**: Create, solve, install, and remove conda packages
+- **Pip package support**: Install and manage pip packages alongside conda packages  
+- **Emscripten/WebAssembly environments**: Built with `emscripten-wasm32` platform in mind
+- **Filesystem operations**: Tools for installing/uninstalling packages into an Emscripten FS
+- **Python environment bootstrapping**: Spawn Python interpreters and load shared libraries
+- **Lock file management**: Generate and manipulate conda lock definitions
+
+Backed by [rattler](https://github.com/conda-incubator/rattler) for conda package solving and an in-house PyPi package manager.
+
+## What Mambajs is NOT (yet)
+
+- **Not a replacement** for micromamba or rattler CLIs
+- **Lock file focused**: The `install`/`remove`/`pipUninstall` functions manipulate conda lock definitions (lock input → lock output). They don't actually install/uninstall packages - it's up to you to handle the actual package installation where and how you want.
+
+## Installation
+
+```bash
+npm install @emscripten-forge/mambajs
+```
+
+Or with yarn:
+
+```bash
+yarn add @emscripten-forge/mambajs
+```
+
+## Quick Start
+
+### Basic Environment Creation
+
+```typescript
+import { create } from '@emscripten-forge/mambajs';
+
+const yml = `
+channels:
+  - https://prefix.dev/emscripten-forge-dev  
+  - https://prefix.dev/conda-forge
+dependencies:
+  - python=3.11
+  - numpy
+  - pandas
+  - pip:
+    - matplotlib
+    - ipywidgets
+`;
+
+// Create environment from YAML definition
+const lock = await create(yml);
+console.log('Created environment with packages:', Object.keys(lock.packages));
+```
+
+### Solving Dependencies
+
+```typescript
+import { solve } from '@emscripten-forge/mambajs';
+
+// Solve for specific packages
+const result = await solve({
+  ymlOrSpecs: ['python=3.11', 'numpy', 'pandas'],
+  logger: console // Optional logger
+});
+
+// Solve with pip packages
+const resultWithPip = await solve({
+  ymlOrSpecs: ['python=3.11'],
+  pipSpecs: ['matplotlib', 'ipywidgets==8.0.0'],
+  logger: console
+});
+```
+
+### Managing Existing Environments
+
+```typescript
+import { install, remove, pipInstall, pipUninstall } from '@emscripten-forge/mambajs';
+
+// Install new conda packages
+const updatedEnv = await install(
+  ['scipy', 'matplotlib'], // packages to install
+  currentLock,              // current environment lock
+  ['conda-forge'],          // additional channels (optional)
+  console                   // logger (optional)
+);
+
+// Remove conda packages  
+const envAfterRemoval = await remove(
+  ['scipy'],    // packages to remove
+  updatedEnv,   // current environment
+  console       // logger (optional)
+);
+
+// Install pip packages
+const envWithPip = await pipInstall(
+  ['requests>=2.28.0', 'flask'],
```

Comment:
flask is not a good example since you probably won't be able to use it in a WASM env. Replace it with 'bqplot' for example

```suggestion
  ['requests>=2.28.0', 'bqplot'],
```